File: app/src/main/java/mx/edu/upqroo/kristenandroid/viewModels/ContactViewModel.java
package mx.edu.upqroo.kristenandroid.viewModels;

import android.app.Application;

import java.util.List;

import androidx.annotation.NonNull;
import androidx.lifecycle.AndroidViewModel;
import androidx.lifecycle.LiveData;
import mx.edu.upqroo.kristenandroid.data.database.entities.Contact;
import mx.edu.upqroo.kristenandroid.data.repositories.ContactRepository;

public class ContactViewModel extends AndroidViewModel {
    private ContactRepository mRepo;

    public ContactViewModel(@NonNull Application application) {
        super(application);
        mRepo = ContactRepository.getInstance(application);
    }

    public LiveData<List<Contact>> getContacts() {
        return mRepo.getContact();
    }

    public void updateFromService() {
        mRepo.updateContacts();
    }
}

File: app/src/main/java/mx/edu/upqroo/kristenandroid/viewModels/ContactViewModel.kt
package mx.edu.upqroo.kristenandroid.viewModels

import android.app.Application
import androidx.lifecycle.AndroidViewModel
import androidx.lifecycle.LiveData
import mx.edu.upqroo.kristenandroid.data.database.entities.Contact
import mx.edu.upqroo.kristenandroid.data.repositories.ContactRepository

class ContactViewModel(application: Application) : AndroidViewModel(application) {
    private val mRepo: ContactRepository = ContactRepository.getInstance(application)

    val contacts: LiveData<List<Contact>>
        get() = mRepo.contact

    fun updateFromService() {
        mRepo.updateContacts()
    }
}
